### ISSUE TITLE
Why scale the weight_decay using 'batch_size * accumulate / nbs' in train.py

### ISSUE THREAD

**Issue by HanxSmile** [ultralytics/yolov5#6757]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I noticed before creating the optimizers, yolo5 calculate the accumulate times and  then scale the `hyp['weight_decay']` using `batch_size * accumulate / nbs` in train.py. Does that mean when using larger batch size, larger weight decay  value helps? If so, is there any theory evidence? or I miss sth important.  Cound any kind one explain my confusion，thanks in advance.

**Comment by glenn-jocher** [comment]
@HanxSmile yes, this allows weight decay and loss magnitude to scale with batch size. This allows YOLOv5 to train at arbitrary batch size with similar results. See the YOLOv5 batch size study for empirical results of the current implementation over batch size.

🌟💡 YOLOv5 Study: mAP vs Batch-Size https://github.com/ultralytics/yolov5/discussions/2452
